src/workers/parser/frameworks.js

```diff
@@ -5,20 +5,22 @@
 import getSymbols from "./getSymbols";
 
 export function isReactComponent(source) {
-  const { imports, classes } = getSymbols(source);
+  const { imports, identifiers, classes } = getSymbols(source);
 
   if (!imports || !classes) {
     return false;
   }
 
-  return importsReact(imports) && extendsComponent(classes);
+  return importsReact(imports, identifiers) || extendsComponent(classes);
 }
 
-function importsReact(imports) {
-  return imports.some(
-    importObj =>
-      importObj.source === "react" &&
-      importObj.specifiers.some(specifier => specifier === "React")
+function importsReact(imports, identifiers) {
+  return (
+    imports.some(
+      importObj =>
+        importObj.source === "react" &&
+        importObj.specifiers.some(specifier => specifier === "React")
+    ) || identifiers.some(identifier => identifier.name === "React")
```

Comment:
this is an interesting change. Why is this needed? Also, could we add some new tests that would fail w/o it

Reply:
I bet this change could be in its own PR actually.

Reply:
This is what caused the problem I mentioned earlier:
>  for some reason isReactComponent was false the whole evening for me even with the current code

`import React` was expected here and it didn't trigger by `const React = require('react')`, so I added the identifier check as well

Reply:
Ahh brilliant. Lets do this in a separate PR. It fixes another problem :)

Reply:
Actually, #4737 is already working on it

Reply:
Oh, didnt realize that
I just kinda did it cause it was interfering with what I was doing :D

Reply:
Yep! I do that all the time too :)

glad you got to see the parser code. It's my favorite part of the codebase